src/rapids_dependency_file_generator/schema.json

```diff
@@ -0,0 +1,160 @@
+                                "additionalProperties": false
+                            }
+                        },
+                        "specific": {
+                            "type": "array",
+                            "items": {
+                                "type": "object",
+                                "properties": {
+                                    "output_types": {"$ref": "#/$defs/Outputs"},
+                                    "matrices": {"$ref": "#/$defs/Matrices"}
+                                },
+                                "required": ["output_types", "matrices"],
+                                "additionalProperties": false
+                            }
+                        }
+                    },
+                    "minProperties": 1,
+                    "additionalProperties": false
+                }
+            }
+        },
+        "channels": {"$ref": "#/$defs/Channels"}
+    },
+    "required": ["files", "dependencies"],
+    "additionalProperties": false,
+    "$defs": {
+        "Channel": {
+            "type": "string",
+            "format": "iri-reference"
+        },
+        "ChannelList": {
+            "type": "array",
+            "items": {
+                "$ref": "#/$defs/Channel"
+            }
+        },
+        "Channels": {
+            "$oneOf": [
+                {"$ref": "#/$defs/Channel"},
+                {"$ref": "#/$defs/ChannelList"}
+            ]
+        },
+        "Matrix": {
+            "type": "object",
+            "patternProperties": {
+                ".*": {
+                    "type": "array",
+                    "items": {"type": "string"}
+                }
+            }
+        },
+        "MatrixMatcher": {
+            "type": "object",
+            "properties": {
+                "matrix": {
+                    "oneOf": [
+                        {
+                            "type": "object",
+                            "patternProperties": {
+                                ".*": {"type": "string"}
+                            }
+                        },
+                        {"type": "null"}
+                    ]
+                },
+                "packages": {"oneOf": [
+                    {"$ref": "#/$defs/Requirements"},
+                    {"type": "null"}
+                ]}
+            },
+            "requiredProperties": ["matrix", "packages"],
+            "additionalProperties": false
+        },
+        "Matrices": {
+            "type": "array",
+            "items": {"$ref": "#/$defs/MatrixMatcher"}
+        },
+        "OutputTypes": {
+            "enum": ["conda", "requirements"]
+        },
+        "OutputList": {
+            "type": "array",
+            "item": {"$ref": "#/$defs/OutputTypes"}
+        },
+        "Outputs": {
+            "oneOf": [
+                {"$ref": "#/$defs/OutputTypes"},
+                {"$ref": "#/$defs/OutputList"},
+                {"const": "none"}
```

Comment:
I like this separation. It didn't initially strike me that we could (or should) be preventing none from being in the list of outputs, using an enum with this constant separated is nice.